common/k8s/src/middleware/metadata.rs

```diff
@@ -82,56 +105,51 @@ impl K8sMetadata {
         };
 
         let watched = StreamBackoff::new(watcher(api, params), ExponentialBackoff::default());
+        let swap_intent = StoreSwapIntent::new(self, store.clone());
 
-        let reflector = reflector(store_writer, watched);
-
-        let driver = {
-            let store = store.clone();
-            async move {
-                reflector
-                    .filter_map(|r| async {
-                        match r {
-                            Ok(event) => Some(event),
-                            Err(e) => {
-                                warn!("k8s watch stream error: {}", e);
-                                None
-                            }
-                        }
+        Ok((
+            reflector(store_writer, watched)
+                .map_ok(|event| {
+                    event.modify(|pod| {
+                        pod.managed_fields_mut().clear();
                     })
-                    .for_each(|p| async {
+                })
+                .inspect(move |event| match event {
+                    Ok(p) => {
                         K8sMetadata::handle_pod(&store, p)
                             .unwrap_or_else(|e| warn!("unable to process pod event: {}", e));
-                    })
-                    .await
-            }
-        };
-
-        Ok((Box::pin(driver), K8sMetadata { store }))
+                    }
+                    Err(e) => warn!("k8s watch stream error: {}", e),
+                })
+                .take_while(|x| std::future::ready(x.is_ok()))
+                .map(Result::unwrap),
```

Comment:
safe unwrap?

Reply:
Yes, `take_while` will terminate the stream if an error is encountered, so the unwrap will never get an error.